### PR TITLE
Added inventory concept docs, along with corresponding schema diagram

### PR DESCRIPTION
**Pull request by jimbaker**

### CODE REVIEW

File: doc/source/high-level-design.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Users
+    Configure and query inventory. The pattern of usage is top-down,
+    whether that's configuring a specific label or drilling down from
+    a given cell.
+
+    Users also can run workflows. This capability implies that
+    workflows can be linked to roles; and that permissions include
+    being the ability to run workflows.
+
+Principals
```
Comment (sigmavirus24) [inline doc/source/high-level-design.rst]: Is this meant to dangle here?

Comment (jimbaker) [inline doc/source/high-level-design.rst]: Thanks for catching this dangling term. I'm sure my intent was to expand more on principals, but obviously it didn't happen in this change!

**Comment by sulochan** [comment]
LGTM

**Comment by sigmavirus24** [comment]
Looks good to me a swell

**Pull request merged**